docs/site/Creating-components.md

```diff
@@ -33,13 +33,13 @@ The example `MyComponent` above will add `MyController` to application's API and
 
 ## Providers
 
-Providers enable components to export values that can be used by the target application or other components. The `Provider`  class provides a `value()` function called by [Context](Context.md) when another entity requests a value to be injected.
+Providers enable components to export values that can be used by the target application or other components. The `Provider` class provides a `value()` function called by [Context](Context.md) when another entity requests a value to be injected.
 
-```js
+```ts
 import {Provider} from '@loopback/context';
 
-export class MyValueProvider {
-  value() {
+export class MyValueProvider implements Provider<string> {
+  value(): string {
```

Comment:
`: string` is needed as now TS can infer the return type of `value()` as `MyValueProvider implements Provider<string>`.